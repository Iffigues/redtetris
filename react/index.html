<!doctype html>
<html>
  <head>
    <title>RedTetris</title>
    <link rel="icon" href="http://redpelicans.com/favicon.ico">                                                                                                                                          
  </head>
  <body style="margin: 0">
    <div id="tetris"></div>
    <script src="/bundle.js"></script>
  </body>
</html>
