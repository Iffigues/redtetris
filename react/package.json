{
  "name": "red_tetrisboilerplate",
  "version": "0.0.1",
  "author": "redpelicans",
  "license": "MIT",
  "scripts": {
    "eslint": "eslint src",
    "client-dist": "NODE_ENV=production webpack --progress --colors",
    "client-dev": "webpack-dev-server --colors --hot --inline --host 0.0.0.0 --port 8080",
    "srv-dev": "DEBUG=tetris:* babel-watch -w src src/server/main.js",
    "srv-dist": "DEBUG=tetris:* babel src --out-dir dist",
    "test": "mocha --require babel-core/register --reporter spec",
    "coverage": "NODE_ENV=test nyc --include src --all -r text -r lcov -r text --require babel-core/register mocha "
  },
  "dependencies": {
    "@babel/preset-es2017": "^7.0.0-beta.53",
    "@babel/preset-react": "^7.10.4",
    "eslint-config-airbnb": "^18.2.0",
    "react": "^15.0.2",
    "react-dom": "^15.0.2",
    "react-hot-loader": "^4.12.21",
    "react-redux": "^4.4.5",
    "redux": "^3.5.2",
    "redux-logger": "^2.6.1",
    "redux-thunk": "^2.0.1",
    "socket.io": "^2.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.10.5",
    "@babel/preset-env": "^7.10.4",
    "babel-cli": "^6.7.7",
    "babel-core": "^6.7.7",
    "babel-eslint": "^6.1.0",
    "babel-loader": "^8.1.0",
    "babel-plugin-__coverage__": "^0.111111.1",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-react": "^6.5.0",
    "babel-preset-stage-0": "^6.5.0",
    "babel-watch": "^2.0.2",
    "chai": "^3.5.0",
    "chai-as-promised": "^5.3.0",
    "chai-equal-jsx": "^1.0.9",
    "eslint": "^4.18.2",
    "eslint-plugin-babel": "^3.3.0",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-react": "^5.2.2",
    "mocha": "^8.0.1",
    "nyc": "^15.1.0",
    "webpack": "^4.43.0",
    "webpack-cli": "^3.3.12",
    "webpack-dev-middleware": "^1.6.1",
    "webpack-dev-server": "^3.11.0",
    "webpack-hot-middleware": "^2.10.0"
  },
  "nyc": {
    "exclude": [
      "build"
    ]
  }
}
