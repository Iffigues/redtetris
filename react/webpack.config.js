const path = require('path');

module.exports = {
  entry: {
    app: './src/client/index.js'
  },
  output: {
    filename: 'bundle.js',
    path: path.resolve(__dirname, 'dist')
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        exclude: /node_modules/,
        use: [{
          loader: 'babel-loader',
          options: {
              cacheDirectory: true,
              babelrc: false,
              presets: [["@babel/env"], "@babel/react"],
              plugins: ["react-hot-loader/babel"]
          }
        }]
      }
    ]
  }
};
