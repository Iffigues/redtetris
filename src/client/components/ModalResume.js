// Libs
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import _ from 'lodash'


const useStyles = makeStyles((theme) => ({
  modal: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  paper: {
    backgroundColor: theme.palette.background.paper,
    border: '2px solid #000',
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
  }
}));

const ModalResume = ({ setSong, isPlaying, song, player, uuidRoom }) => {
  const classes = useStyles();

  const handleCloseModal = () => {
    console.log("HandleCloseModal")
  };

  const leaveRoom = (e) => {
    e.preventDefault()
    sendSocket('server/leave-room', { uuidRoom, uuidUser: player.uuid })
  }

  const resume = () => {
    sendSocket('server/pause-resume', { channel: uuidRoom })
  }

  const changeSongPref = (e) => {
    e.preventDefault()
    setSong(!song)
  }

  return (
    <div className="test--modal-resume">
      <Modal
        className="d-flex jcnt--center aitems--center fdir--row pt-3"
        open={isPlaying}
        disablePortal
        disableEnforceFocus
        disableAutoFocus
        onClose={handleCloseModal}
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
      >
        <div className={classes.paper}>
          <div className="aself--center">
            <h1 className="aself--center" id="transition-modal-title">Pause</h1>
          </div>
          <div className="d-flex jcnt--space-ar fdir--row">
            <Button
              id="songRoom"
              data-testid='btnSong'
              onClick={e => changeSongPref(e)}
            >
              { song ? "ðŸ”ˆ" : "ðŸ”‡" }
            </Button>
          </div>
          <div className="d-flex jcnt--space-ar fdir--row">
            <div className="aself--fstart p-2">
              <Button
                id="leaveRoom"
                variant="contained"
                color="secondary"
                data-testid='btnCreateRoom'
                onClick={e => leaveRoom(e)}
              >
                Quitter la partie ?
              </Button>
            </div>
            <div className="aself--fstart p-2">
              <Button
                id="resume"
                variant="contained"
                color="primary"
                data-testid='btnCreateRoom'
                onClick={resume}
              >
                Reprendre
              </Button>
            </div>
          </div>
        </div>
      </Modal>
    </div>
  )
}

export default ModalResume