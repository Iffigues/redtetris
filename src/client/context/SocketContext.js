import React, { createContext } from "react"
import { params } from "../../../params"
import socketIOClient from "socket.io-client"

export const SocketContext = createContext();

export const SocketContextProvider = ({ children }) => {
  const socketClient = socketIOClient(params.url);  
  const sendSocket = (type, data = false) => {
    socketClient.emit(type, data);
  }
  
  return (
    <SocketContext.Provider
      value={{ socketClient, sendSocket }}
    >
      { children }
    </SocketContext.Provider>
  )
}