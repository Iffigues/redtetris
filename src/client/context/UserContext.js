import createDataContext from "./createDataContext";

const initialState = {
  uuidRoom: null,
  uuidUser: null
};

const UserReducer = (state, payload) => {
  switch (payload.type) {
    case "updateUuidRoom":
      return {  ...state, uuidRoom: payload.uuidRoom };
    case "updateUuidUser":
      return {  ...state, uuidUser: payload.uuidUser };
    default:
      return state;
  }
};

const updateUuidRoom = dispatch => (uuidRoom) => 
  dispatch({ type: "updateUuidRoom", payload: uuidRoom });

const updateUuidUser = dispatch => uuidUser => 
  dispatch({ type: "updateUuidUser", payload: uuidUser });

export const { Provider, Context } = createDataContext(
  //reducer :
  UserReducer,

  //action functions :
  {
    updateUuidRoom,
    updateUuidUser
  },

  //initialState :
  {
   ...initialState
  }
);