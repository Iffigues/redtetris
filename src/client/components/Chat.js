import React, { useState, useContext } from 'react';
import { Button, Input } from '@material-ui/core';

import { Context as RoomsContext } from "../context/RoomsContext";
import { SocketContext } from "../context/SocketContext";
import { Context as UserContext } from "../context/UserContext";

const Chat = ({ uuidRoom }) => {
  const { sendSocket } = useContext(SocketContext);
  const { state: { player } } = useContext(UserContext);
  const { state: { rooms } } = useContext(RoomsContext);
  const [message, setMessage] = useState('');
  // const [playingAudio, setplayingAudio] = useState(true);

  const sendMessage = (e) => {
    e.preventDefault();
    if (message !== "") {
      sendSocket('server/new-message', {
          content: message,
          uuidRoom,
          login: player.name,
          id_user: player.uuid
      })
      setMessage("")
      const element = document.getElementById("chat");
      element.scrollTop = 0;
    }
  }

return (
  <div>
    <div id="chat" className="container--chat">
      <div className="chat d-flex fdir--column">
        {
            rooms && rooms[uuidRoom].messages.map(msg => {
              return (
                <div
                  className={`${msg.uuidUser === player.uuid
                              ? 'aself--fend bubble--me'
                              : 'aself--fstart bubble--other'}
                              aself--fstart
                              pb-2
                              bubble
                            `}
                >
                  <p>
                    <span className="bold medium-text">{ msg.login }:</span> { msg.content } <br />
                  </p>
                  <p className="thin small-text pt-0">{msg.time}</p>
                </div>
              )
            })
          }
        </div>
      </div>
      <div className="d-flex fdir--column">

      <div className="aself--center">
        <form onSubmit={sendMessage}>
            <Input
              className="aself--center mb-2 input-size"
              value={message}
              onChange={e => setMessage(e.target.value)}
              variant="outlined"
              required
              name="message"
              label="message"
              type="message"
              id="message"
              inputProps={{
                'data-testid': 'messageChat'
              }}
              />
            <Button
              className="aself--center mt-2 test--btn-join-room"
              color="primary"
              data-testid='btnLogin'
              disabled={message.length === 0}
              onClick={sendMessage}
              >
              ➡️
            </Button>
          </form>
        </div>
      </div>
    </div>
  )
}

export default Chat
