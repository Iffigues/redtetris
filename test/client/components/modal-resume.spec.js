import React, { useContext, useEffect } from "react";
import '@testing-library/jest-dom/extend-expect';
import Enzyme, { mount, shallow } from "enzyme";
import Chat from '../../../src/client/components/Chat'
import { render, fireEvent } from '@testing-library/react'
import Adapter from "enzyme-adapter-react-16";
import { player_instance1, room1, visitor_player, rooms_1 } from '../helpers/data'
import { TestAppSocketProvider } from "../helpers/socketContext";
import { TestAppUserProvider } from "../helpers/userContext";
import { TestAppRoomsProvider } from "../helpers/roomsContext";
import { Context as UserContext } from "../../../src/client/context/UserContext";
import { Context as RoomsContext } from "../../../src/client/context/RoomsContext";
import ModalResume from "../../../src/client/components/ModalResume";


Enzyme.configure({ adapter: new Adapter() });
describe('ModalResume component', () => {

  const Wrapper = () => (
    <ModalResume
      setSong={() => {}}
      isPlaying={false}
      song={false}
      player={player_instance1}
      uuidRoom={room1.channel}
    />
  )

  it('Is exists', () => {
    const wrapper = shallow(<Wrapper />);
    expect(wrapper).toMatchSnapshot();
  });


  it('Can mount', () => {
    render(<Wrapper />);
  })

  it('is not null', () => {
    const { container } = render(<Wrapper />);
    const container_modal = container.querySelector('.test--modal-resume')
    expect(container_modal).not.toBeNull()
  })

});