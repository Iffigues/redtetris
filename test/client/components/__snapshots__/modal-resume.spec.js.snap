// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalResume component Is exists 1`] = `
<ModalResume
  isPlaying={false}
  player={
    Player {
      "action": Object {
        " ": [Function],
        "ArrowDown": [Function],
        "ArrowLeft": [Function],
        "ArrowRight": [Function],
        "ArrowUp": [Function],
      },
      "addSheet": [Function],
      "addSheetFunc": [Function],
      "admin": true,
      "block": null,
      "blockCPY": [Function],
      "canPose": [Function],
      "cantPose": false,
      "changePlaying": [Function],
      "copyBlock": [Function],
      "createIntervalGame": [Function],
      "currentMapGame": Array [
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      ],
      "destroyFunc": [Function],
      "destroyLine": [Function],
      "down": [Function],
      "draw": [Function],
      "end": false,
      "fallen": [Function],
      "indestructible": 0,
      "initActionObject": [Function],
      "initGame": [Function],
      "interval": 1000,
      "isPlaying": false,
      "left": [Function],
      "lock": true,
      "lrs": Array [
        Array [
          0,
          -1,
        ],
        Array [
          1,
          0,
        ],
      ],
      "makeShadow": [Function],
      "move": [Function],
      "name": "owalid1",
      "nextMapGame": Array [
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      ],
      "pushSheet": [Function],
      "requestNewGame": false,
      "retro": [Function],
      "right": [Function],
      "rotate": [Function],
      "rotateL": [Function],
      "rotateR": [Function],
      "score": 0,
      "sendMap": [Function],
      "setMoose": [Function],
      "setRequestNewGame": [Function],
      "setVisitor": [Function],
      "shadow": null,
      "sheets": Array [],
      "space": [Function],
      "srs": Array [
        Array [
          0,
          1,
        ],
        Array [
          -1,
          0,
        ],
      ],
      "startGame": [Function],
      "time": 1000,
      "timing": 1000,
      "updateRoomFunction": [Function],
      "uuid": "26ca3915-7c5f-423e-a605-83f61e281198",
      "verifLine": [Function],
      "visitor": false,
      "wash": [Function],
      "willBePosed": [Function],
    }
  }
  setSong={[Function]}
  song={false}
  uuidRoom="5f64d170-87bf-4094-9363-22eb743a0ac7"
/>
`;
